extractor/src/main/scala/com/digitalasset/extractor/ledger/types/LedgerValue.scala

```diff
@@ -87,15 +85,25 @@ object LedgerValue {
   private def convertOptional(apiOptional: api.value.Optional) =
     apiOptional.value traverseU (_.convert) map (V.ValueOptional(_))
 
-  private def convertMap(apiMap: api.value.Map): String \/ OfCid[V.ValueTextMap] =
+  private def convertTextMap(apiMap: api.value.Map): String \/ OfCid[V.ValueTextMap] =
     for {
       entries <- apiMap.entries.toList.traverseU {
-        case api.value.Map.Entry(_, None) => -\/("value must be defined")
         case api.value.Map.Entry(k, Some(v)) => v.sum.convert.map(k -> _)
+        case api.value.Map.Entry(_, None) => -\/("value must be defined")
```

Comment:
I know that you are not changing the error here, but it would be nice if we added some context. Can we add the key for which the value is not defined?